### ISSUE TITLE
GitLens not working for some files in the repo, log error: 'foo.txt' is outside repository  Failed with exit code: 128

### ISSUE THREAD

**Issue by kiewic** [eamodio/vscode-gitlens#105]
- GitLens Version: 4.2.0
- VSCode Version: 1.13.1
- OS Version: Windows 10 Version 1703 (Build 15063)
- Git version: 2.12.1.windows.1

GitLens work for some files in my repository, but it does not work for other files.

The GitLens logs show there are errors for the files where it is not working:

```plain
git -c core.quotepath=false ls-files c:\Repos\myProject\bar\foo.ts   cwd='c:/Repos/myProject/'
  Error: fatal: c:\Repos\myProject\bar\foo.ts: 'c:\Repos\myProject\bar\foo.ts' is outside repository  Failed with exit code: 128
```

The same command works for the files where GitLens is working fine:

```plain
git -c core.quotepath=false ls-files c:\repos\myProject\buzz\foo.ts   cwd='c:/repos/myProject/'
```

The issue seems to be the capital letter in `Repos`. When `Repos` is capitalized, GitLens does not work, but when it is all in lower case, it works.

**Comment by eamodio** [comment]
@kiewic I'm not 100% sure it is this issue: https://github.com/Microsoft/vscode/issues/12448 but it certainly sounds like it. The workaround is to make sure you open vscode to the folder path with the exact casing as on your file system otherwise you can run into all kinds of odd issues in vscode and GitLens.

Hope that helps.

**Comment by kiewic** [comment]
@eamodio It sounds like the same issue, although I had never faced this problem until I started using vscode-gitlens.<sup>1</sup> The tricky part is that the issue repros for some files in the same instance of VSCode while other files work just fine. Apparently the case mixing does not come only from the VSCode open folder path, it could also come by typing different casings in the open file dialog or directly into the command bar. Interestingly, once the file has been queried successfully by vscode-gitlens once, git-lens displays its information correctly the rest of the times, even if it is opened again with a "wrong" casing. Could the issue maybe be in `git ls-files` for not ignoring casing in case-insensitive file systems?

<sup>1</sup> BTW, thanks for the awesome extension!

**Comment by eamodio** [comment]
@kiewic Yeah, I've looked into this a bunch (some more details here https://github.com/Microsoft/vscode/issues/26187). The crux of the issue is that vscode doesn't normalize the paths to be the actual path (in the file system), which creates an issue for case-sensitive tools (like git). I've tried specific changes in GitLens to deal with it, but nothing panned out (especially since there is no great way right now to get the "real" path of a file). At this point, I'm relying on vscode to fix the root issue or users to open files with the real casing :(
